### PR TITLE
Refactor a bunch of tests to use template literal syntax instead of [].join(

### PR DESCRIPTION
**Pull request by jseminck**
I wanted to experiment writing a code mod and decided to write one to transform the `code: [].join('\n');` syntax to use template literals instead.

I ran it on a file by file basis. I could not transform all the files yet because template literals introduce the following issues:

* Error lines / columns are different. Since my syntax adds 1 extra line + a lot of indentation, those numbers change
* Output often also doesn't work because of the added indentation in the code through the template literal sting

But a lot of files didn't rely on this so at least we could start with transforming those 😄 

Here's the codemod code: https://astexplorer.net/#/gist/000b6e8110a245ca6384c260b954e452/1687ee893085fdae7119c809294e5ed43a3fe428

### CODE REVIEW

File: tests/lib/rules/boolean-prop-naming.js [hunk 1]
```diff
@@ -91,222 +86,205 @@ ruleTester.run('boolean-prop-naming', rule, {
...
+    code: `
+      var Hello = ({isSomething}) => { return <div /> }
+      Hello.propTypes = {isSomething: PropTypes.bool};
+    `,
     options: [{
       rule: '^is[A-Z]([A-Za-z0-9]?)+'
     }],
     parser: 'babel-eslint'
   }, {
-    // Functional components and Flowtype booleans
```
Comment (ljharb) [inline tests/lib/rules/boolean-prop-naming.js]: looks like your codemod removed a bunch of comments. please restore them all

Comment (jseminck) [reply tests/lib/rules/boolean-prop-naming.js]: Woops! My bad... I found the mistake 😄 I'll re-add all of them. I think most are actually in this rule where every test has a comment.

**Comment by ljharb** [comment]
@jseminck would you mind one more rebase and rerun? Please ping me when its done and I'll merge asap. Thanks!

**Comment by LestaD** [comment]
@ljharb @jseminck  maybe squash and merge?

**Comment by ljharb** [comment]
@LestaD a) the issue is with staleness, not with "i only want one commit", b) github's squashmerge and rebase merge are terrible features that pollute the git log; rebasing PRs and using a fast-forwardable merge commit is far superior.

**Comment by jseminck** [comment]
I don't think any changes have been done in master that affect this. Will check!

**Pull request merged**